packages/backend-function/src/function_env_type_generator.ts

```diff
@@ -69,10 +77,15 @@ export class FunctionEnvironmentTypeGenerator {
   }
 
   /**
-   * Generate an un-typed process.env shim
+   * Generate an any-typed process.env shim if doesn't exist
    */
   generateProcessEnvShim = () => {
-    const content = `${this.header}${EOL}${this.envAssignment};`;
+    // Don't recreate the file
+    if (fs.existsSync(this.typeDefFilePath)) {
+      return;
+    }
+
+    const content = `${this.header}${EOL}${this.envPreAssignment};`;
```

Comment:
nit: we could get rid of `envPreAssignment` and use `envAssignment` here as `${this.envAssignment} as any;`